test/test_s3_utils.py

```diff
@@ -339,35 +410,50 @@ def test_s3utils_get_google_key():
 @pytest.mark.unit
 @using_fresh_ff_state_for_testing()
 def test_s3utils_get_access_keys_with_old_style_default():
-    util = s3Utils(env='fourfront-mastertest')
-    with mock.patch.object(util, "get_key") as mock_get_key:
-        actual_key = {'key': 'some-key', 'server': 'some-server'}
+    s3u = s3Utils(env='fourfront-mastertest')
+    with mock.patch.object(s3u, "get_key") as mock_get_key:
 
         def mocked_get_key(keyfile_name):
             ignored(keyfile_name)
             key_wrapper = {'default': actual_key}
             return key_wrapper
 
         mock_get_key.side_effect = mocked_get_key
-        key = util.get_access_keys()
+
+        actual_key = {'secret': 'some-secret', 'server': 'some-server'}
+
+        with pytest.raises(ValueError):
+            s3u.get_access_keys()  # ill-formed secret is missing key. portal keys are expected to have a key.
+
+        key = s3u.get_access_keys(require_key=False)  # accesses to jupyterhub keys could need this
+        assert key == actual_key
```

Comment:
This test illustrates use of the new `require_key=False` arg. I did this because I _think_ I caught all the references we need to update, but if not, any code that gets broken can just add this argument and should be fine. I actually think anything expecting a missing key is using `.get_key()` not `.get_access_keys()`.